### ISSUE TITLE
Command "haskell-language-server-wrapper --lsp -d -l /tmp/hls.log" is not present on the path.

### ISSUE THREAD

**Issue by mpardalos** [emacs-lsp/lsp-mode#3507]
### Thank you for the bug report

- [X] I am using the latest version of `lsp-mode` related packages.
- [X] I checked [FAQ](https://emacs-lsp.github.io/lsp-mode/page/faq/) and [Troubleshooting](https://emacs-lsp.github.io/lsp-mode/page/troubleshooting/) sections
- [ ] You may also try reproduce the issue using clean environment using the following command: `M-x lsp-start-plain`

### Bug description

Starting the lsp server in a haskell project gives the following error in *lsp-log*:

`Command "haskell-language-server-wrapper --lsp -d -l /tmp/hls.log" is not present on the path.`

I tried lsp-start-plain, I don't get the same message, but the language server still does not start.

I also tried running `haskell-language-server-wrapper` manually in the project directory and it runs correctly.
It looks to me like lsp is running "haskell-language-server-wrapper --lsp -d -l /tmp/hls.log" as one command, without correctly separating each argument.

### Steps to reproduce

I am using doom-emacs. 

Simply open a file in a haskell project. Running `M-x lsp` should also not start the server and print the message again.

### Expected behavior

Opening a haskell file should open the language server.

### Which Language Server did you use?

lsp-haskell (haskell-language-server)

### OS

Linux

**Comment by mpardalos** [comment]
This turned out to be a PATH issue. I am in WSL and was launching emacs through the shortcut, meant it was not getting the PATH from .bashrc. Closing.

**Comment by evincarofautumn** [comment]
I ran into the same issue on macOS because I was launching `emacs --daemon` at startup with `launchctl`, so it was getting the default `PATH`.

> It looks to me like lsp is running "haskell-language-server-wrapper --lsp -d -l /tmp/hls.log" as one command, without correctly separating each argument.

I spent some time chasing the same thing. So maybe the message should be something like “in command ⟨*full command*⟩, ⟨*executable name*⟩ wasn’t found on `PATH`” to indicate that what’s lacking from `PATH` is just `haskell-language-server-wrapper`, not the whole command.
